src/commandDetails/coin/leaderboard.ts

```diff
@@ -0,0 +1,72 @@
+import { container, SapphireClient } from '@sapphire/framework';
+import { Client, MessageEmbed } from 'discord.js';
+import {
+  CodeyCommandDetails,
+  CodeyCommandResponseType,
+  getUserIdFromMessage,
+  SapphireMessageExecuteType,
+  SapphireMessageResponse
+} from '../../codeyCommand';
+import { getCurrentCoinLeaderboard, getCoinBalanceByUserId, UserCoinEntry } from '../../components/coin';
+import { EMBED_COLOUR } from '../../utils/embeds';
+
+const getCurrentCoinLeaderboardEmbed = async (
+  client: SapphireClient<boolean>,
+  leaderboard: UserCoinEntry[],
+  currentUserId: string,
+  limit = 10
+): Promise<MessageEmbed> => {
+  const currentLeaderboardEmbed = new MessageEmbed().setColor(EMBED_COLOUR).setTitle('Current CodeyCoin Leaderboard');
+  // Initialise user's coin balance if they have not already
+  const userBalance = await getCoinBalanceByUserId(currentUserId);
+  const currentPosition = leaderboard.findIndex((userCoinEntry) => userCoinEntry.user_id === currentUserId) + 1;
+
+  let currentLeaderboardText = '';
+  for (let i = 0; i < limit; i++) {
+    if (leaderboard.length <= i) break;
+    const userCoinEntry = leaderboard[i];
+    const user = await client.users.fetch(userCoinEntry.user_id);
+    const userTag = user?.tag ?? '<unknown>';
+    const userCoinEntryText = `${i + 1}. ${userTag} - ${userCoinEntry.balance} 🪙.\n`;
+    currentLeaderboardText += userCoinEntryText;
+  }
+  currentLeaderboardEmbed.addFields(
+    {
+      name: 'Current Leaderboard',
```

Comment:
Can we remove this heading (I think it might seem a bit redundant)? I think the contents of the leaderboard can probably go into the embed description.

Reply:
Alright this is now done